x-pack/plugins/alerting/server/alert/create_alert_factory.ts

```diff
@@ -149,8 +150,8 @@ export function createAlertFactory<
             hasReachedAlertLimit,
             alertLimit: maxAlerts,
             autoRecoverAlerts,
-            // setFlapping is false, as we only want to use this function to get the recovered alerts
-            setFlapping: false,
+            // flappingSettings.enabled is false, as we only want to use this function to get the recovered alerts
+            flappingSettings: { enabled: false } as RulesSettingsFlapping,
```

Comment:
This feels like some of the RulesSettingsFlapping should be optional or `Partial<RulesSettingsFlapping>` should be used in ProcessAlertsOpts. There are too much type castings.

Reply:
Resolved in this commit 205ddbbd32d2636fd225a114208dc566fda385d1